### PR TITLE
Scan BarCode API to return NotSupported Error on desktop and web platform

### PR DESCRIPTION
**Pull request by samyuktha24**
ScanBarCode API is only available for mobile platform and not supported on Desktop and web platform
ErrorCode.NOT_SUPPORTED_ON_PLATFORM (i.e 100 errorcode) will be returned in the callback on desktop and web platform.

Tests:
Did manual testing using a custom app, tested the following:

- ScanBarCode API works on mobile platform

- ScanBarCode API returns ErrorCode.NOT_SUPPORTED_ON_PLATFORM (i.e 100 errorcode) on desktop and web platform

### CODE REVIEW

File: src/public/media.ts [hunk 1]
```diff
@@ -437,6 +438,12 @@ export namespace media {
     }
     ensureInitialized(FrameContexts.content, FrameContexts.task);
 
+    if (!GlobalVars.isFramelessWindow) {
+      const notSupportedError: SdkError = { errorCode: ErrorCode.NOT_SUPPORTED_ON_PLATFORM };
+      callback(notSupportedError, undefined);
```
Comment (SonalJha) [inline src/public/media.ts:443]: Please make it null instead of undefined

Comment (samyuktha24) [reply src/public/media.ts]: done.

File: src/public/media.ts [hunk 2]
```diff
@@ -437,6 +438,12 @@ export namespace media {
     }
     ensureInitialized(FrameContexts.content, FrameContexts.task);
 
+    if (!GlobalVars.isFramelessWindow) {
```
Comment (ydogandjiev) [inline src/public/media.ts:441]: This is not a good way to check for web/desktop. Please look at the `GlobalVars.hostClientType` instead or better yet implement a handler in the web/desktop client code itself to return the error.

**Review by ydogandjiev — CHANGES_REQUESTED** [review]

File: src/public/media.ts (hunk 2 above)
Comment (samyuktha24) [reply src/public/media.ts]: Sure. I have modified the code to use GlobalVars.hostClientType.

**Review by ydogandjiev — APPROVED** [review]

**Review by SonalJha — APPROVED** [review]

**Pull request merged**